lib/web_ui/lib/src/engine/view_embedder/display_dpr_stream.dart

```diff
@@ -0,0 +1,55 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:js_interop';
+
+import 'package:ui/src/engine/display.dart';
+import 'package:ui/src/engine/dom.dart';
+import 'package:ui/ui.dart' as ui show Display;
+
+/// Determines if high contrast is enabled using media query 'forced-colors: active' for Windows
+class DisplayDprStream {
```

Comment:
You could also put this class in `engine/display.dart` but it's up to you.

Reply:
I'm not entirely sure where this lives. At first I put it in the `platform_dispatcher` directory, then in the `dimensions_provider` directory... The current one seemed like a good middle ground (close to where it's used). You're right that this now looks more related to the Display than other things, but in that case I'd add a `display` directory. I'll leave it here for now, but yeah, this might not be the best spot!